src_py/camera.py

```diff
@@ -1,82 +1,132 @@
 import os
-import sys
 import platform
+import sys
 import warnings
 from abc import ABC, abstractmethod
 
-_is_init = 0
+from pygame import error
 
+_is_init = False
 
-def _setup_opencv_mac():
-    global list_cameras, Camera, colorspace
 
-    from pygame import _camera_opencv
+class AbstractCamera(ABC):
+    @abstractmethod
+    def __init__(self, device=0, size=(320, 200), mode="RGB"):
+        """ """
 
-    try:
-        from pygame import _camera
-    except ImportError:
-        _camera = None
+    @abstractmethod
+    def start(self):
+        """ """
 
-    list_cameras = _camera_opencv.list_cameras_darwin
-    Camera = _camera_opencv.CameraMac
-    if _camera:
-        colorspace = _camera.colorspace
+    @abstractmethod
+    def stop(self):
+        """ """
 
+    @abstractmethod
+    def get_controls(self):
+        """ """
 
-def _setup_opencv():
-    global list_cameras, Camera, colorspace
+    @abstractmethod
+    def set_controls(self, **kwargs):
+        """ """
 
-    from pygame import _camera_opencv
+    @abstractmethod
+    def get_size(self, dest_surf=None):
+        """ """
 
-    try:
-        from pygame import _camera
-    except ImportError:
-        _camera = None
+    @abstractmethod
+    def query_image(self, dest_surf=None):
+        """ """
+
+    @abstractmethod
+    def get_image(self, dest_surf=None):
+        """ """
+
+    @abstractmethod
+    def get_raw(self):
+        """ """
+
+
+def _pre_init_placeholder(*_, **__):
+    if not _is_init:
+        raise error("pygame.camera is not initialized")
 
-    list_cameras = _camera_opencv.list_cameras
-    Camera = _camera_opencv.Camera
-    if _camera:
-        colorspace = _camera.colorspace
+    # camera was init, and yet functions are not monkey patched. This should
+    # not happen
+    raise NotImplementedError()
```

Comment:
is this supposed to be here?

Reply:
Yeah, just for extra safety, IG. I can't think of a case where this codepath will actually be executed